### PR TITLE
feat: warning modal now able to be disabled

### PR DESCRIPTION
**Pull request by ZigTag**
Fixes #[issue_no]

## Summary of Changes

Added a checkbox to disable the warning modal

## Known Issues
- Modal immediately disappears after clicking checkbox

## Screenshots (if necessary)

image
image
image
image

## Additional context

Discord username (if different from GitHub):

### CODE REVIEW

File: src/renderer/components/GeneralSettings/index.tsx [hunk 1]
```diff
@@ -34,8 +34,23 @@ function InstallPathSettingItem(props: { path: string, setPath: (path: string) =
     );
 }
 
+const DisableWarningSettingItem = (props: {disableWarning: boolean, setDisableWarning: CallableFunction}) => {
+    const handleClick = () => {
+        props.setDisableWarning(!props.disableWarning);
+        settings.set('mainSettings.disableWarning', !props.disableWarning);
```
Comment (nistei) [inline src/renderer/components/GeneralSettings/index.tsx:40]: Race condition.

Setting `disableWarning` first and then using it again. Use a temporary variable.

```suggestion
        const newState = !props.disableWarning;
        props.setDisableWarning(newState);
        settings.set('mainSettings.disableWarning', newState);
```

File: src/renderer/components/WarningModal/index.tsx [hunk 2]
```diff
@@ -1,14 +1,19 @@
-import React from 'react';
```
Comment (nistei) [inline src/renderer/components/WarningModal/index.tsx:1]: The logic for showing the warning should be outside of the component so we can reuse the warning modal for other use-cases.

We shouldn't make a generic `disableWarning` setting. It should be scoped like `disableExperimentalWarning`.

**Review by nistei — CHANGES_REQUESTED** [review]

**Comment by ErickSharp** [comment]
This is glorious and was a pain in the ass for me. Great code.

**Comment by ErickSharp** [comment]
This is also an issue that I noticed when thinking about how to solve this: How do we differentiate between different modals? The messages for either developmental version are different. It seems minor, but what if the user never gets to see the message for the other version after they have selected "dont show again?" Just my thoughts.

File: src/renderer/components/GeneralSettings/index.tsx [hunk 3]
```diff
@@ -34,8 +34,23 @@ function InstallPathSettingItem(props: { path: string, setPath: (path: string) =
...
+    };
+
+    return (
+        <SettingsItem>
+            <SettingItemName>Disable Version Warnings</SettingItemName>
+            <SettingItemContent onClick={() => handleClick()}>{props.disableWarning ? 'Yes' : 'No'}</SettingItemContent>
```
Comment (Benjozork) [inline src/renderer/components/GeneralSettings/index.tsx:46]:
```suggestion
            <SettingItemContent onClick={handleClick}>{props.disableWarning ? 'Yes' : 'No'}</SettingItemContent>
```

**Review by Benjozork — CHANGES_REQUESTED** [review]

**Comment by ZigTag** [comment]
> 
> 
> This is also an issue that I noticed when thinking about how to solve this: How do we differentiate between different modals? The messages for either developmental version are different. It seems minor, but what if the user never gets to see the message for the other version after they have selected "dont show again?" Just my thoughts.

to be completely fair, thats their loss

**Comment by ErickSharp** [comment]
> > This is also an issue that I noticed when thinking about how to solve this: How do we differentiate between different modals? The messages for either developmental version are different. It seems minor, but what if the user never gets to see the message for the other version after they have selected "dont show again?" Just my thoughts.
> 
> to be completely fair, thats their loss

LMAO I think that's more our loss since we can't figure out exactly how to do this...

**Comment by FoxtrotSierra6829** [comment]
> 
> 
> > > This is also an issue that I noticed when thinking about how to solve this: How do we differentiate between different modals? The messages for either developmental version are different. It seems minor, but what if the user never gets to see the message for the other version after they have selected "dont show again?" Just my thoughts.
> > 
> > 
> > to be completely fair, thats their loss
> 
> LMAO I think that's more our loss since we can't figure out exactly how to do this...

I think it should be feasible to just store two different variables, since we've already detected the version for the text.

File: src/renderer/components/GeneralSettings/index.tsx [hunk 4]
```diff
@@ -34,8 +34,24 @@ function InstallPathSettingItem(props: { path: string, setPath: (path: string) =
...
+        </SettingsItem>
+    );
+};
+
 function index(): JSX.Element {
     const [installPath, setInstallPath] = useState<string>(settings.get('mainSettings.msfsPackagePath') as string);
+    const [disableWarning, setDisableWarning] = useState<boolean>(settings.get('mainSettings.disableExperimentalWarning') as boolean);
 
     const handleReset = async () => {
         settings.clear();
```
Comment (FoxtrotSierra6829) [inline src/renderer/components/GeneralSettings/index.tsx:57]: grafik
While waiting for GH access, I'm gonna do it this way... This fixes the state not being displayed correctly after pressing reset

**Review by FoxtrotSierra6829 — CHANGES_REQUESTED** [review]

File: src/renderer/components/GeneralSettings/index.tsx (hunk 4 above)
Comment (ZigTag) [reply src/renderer/components/GeneralSettings/index.tsx:57]: settings.reset handles it automatically, I could change the state though

**Comment by FoxtrotSierra6829** [comment]
LGTM

**Review by FoxtrotSierra6829 — APPROVED** [review]

**Review by FoxtrotSierra6829 — APPROVED** [review]

**Comment by FoxtrotSierra6829** [comment]
Tested, LGTM

**Pull request merged**